src/lib/lib.c

```diff
@@ -7,6 +7,7 @@
 #include <sof/common.h>
 #include <rtos/string.h>
 #include <sof/compiler_info.h>
+#include <xtensa/config/core-isa.h>
```

Comment:
Since this is arch specific include in a generic file, can it be put behind an ifdef for xtos xtnesa configuration ?

Reply:
@cujomalainey Hmm, this should come via compiler_info.h . E.g. in XTOS build this is pulled in for xtensa in
src/arch/xtensa/include/arch/compiler_info.h

In Zephyr, "__XCC__" is passed directly via DEFINES , so we don't need to pull core-isa.h

Reply:
@kv2019i I traced the headers and didnt find the include core-isa in there, but it seems to work without the direct include. Hmm weird

Reply:
```
. /usr/local/google/home/cujomalainey/sof/src/include/sof/common.h                                                                                       
.. /usr/local/google/home/cujomalainey/sof/src/include/sof/debug/panic.h
... /usr/local/google/home/cujomalainey/sof/src/arch/xtensa/include/arch/debug/panic.h
.... /usr/local/google/home/cujomalainey/sof/xtos/include/rtos/cache.h                                                                                         
..... /usr/local/google/home/cujomalainey/sof/src/arch/xtensa/include/arch/lib/cache.h                                                                         
...... /usr/local/google/home/cujomalainey/xtensa/XtDevTools/install/builds/RG-2017.8-linux/cavs2x_LX6HiFi3_2017_8/xtensa-elf/arch/include/xtensa/config/core-isa.h
```

Lol I don't think that is the include path we intended

Reply:
@cujomalainey Heh, I admit that somewhat suprising include path, so indeed better to have this explicitly.